### ISSUE TITLE
HTML 5 input acceptance test may fail if locale doesn't match

### ISSUE THREAD

**Issue by aaltat** [robotframework/SeleniumLibrary#1344]
#1327 introduced tests for the HTML 5 different input types. But because date and time is involved, locale plays an effect and for some users the test may fail. 

Convert date and perhaps time to always same format to fix the issue.

**Comment by rasjani** [comment]
date is still failing for me locally with `ValueError: year is out of range`

and traceback looks like this:


  | Traceback (most recent call last):   File "/Users/jani.mikkonen/src/work/SeleniumLibrary/atest/resources/testlibs/date_parser.py", line 5, in parse_two_dates     return parse(date1), parse(date2)   File "/Users/jani.mikkonen/src/work/SeleniumLibrary/venv2/lib/python2.7/site-packages/dateutil/parser/_parser.py", line 1358, in parse     return DEFAULTPARSER.parse(timestr, **kwargs)   File "/Users/jani.mikkonen/src/work/SeleniumLibrary/venv2/lib/python2.7/site-packages/dateutil/parser/_parser.py", line 654, in parse     ret = self._build_naive(res, default)   File "/Users/jani.mikkonen/src/work/SeleniumLibrary/venv2/lib/python2.7/site-packages/dateutil/parser/_parser.py", line 1229, in _build_naive     naive = default.replace(**repl)
-- | --

Also, input field type time also started to fail after this:


  | 2019-03-20 02:34:00 != 2019-03-20 14:34:00
-- | --

**Comment by aaltat** [comment]
Oh well, I hate different date formats. In what test it fails? Is it this one: https://github.com/robotframework/SeleniumLibrary/blob/master/atest/acceptance/keywords/textfields_html5.robot#L20 ?

**Comment by aaltat** [comment]
By the way, what browser you use run the tests?

**Comment by rasjani** [comment]
chrome & headless chrome - same setup as travis.

First failure: https://github.com/robotframework/SeleniumLibrary/blob/master/atest/acceptance/keywords/textfields_html5.robot#L12

Second failure: https://github.com/robotframework/SeleniumLibrary/blob/master/atest/acceptance/keywords/textfields_html5.robot#L56

**Comment by aaltat** [comment]
Strange dateutil should ensure the correct parsing of the date and time. 

From you python installation, could you do: 
```
>>> from dateutil.parser import parse
>>> d = parse('11/22/2019')
>>> print(d)
2019-11-22 00:00:00
>>> t = parse('02:34PM')
>>> print(t)
2019-03-21 14:34:00
>>>
```
Also it would be nice to see what `parse` methods returns for you when it converts the value in the `${value}` variable. From the `Input field type date` and `Input field type time` and with the value which is returned by the `Get Value` keyword. 

Could you send me the output. Requires [dateutil](https://dateutil.readthedocs.io/en/stable/) is installed.

**Comment by rasjani** [comment]
* Input field type date:


Issue in it  is not about converting but either or both: setting and fetching the value.
This line:
https://github.com/robotframework/SeleniumLibrary/blob/master/atest/acceptance/keywords/textfields_html5.robot#L14 

sets the date to the field.

and https://github.com/robotframework/SeleniumLibrary/blob/master/atest/acceptance/keywords/textfields_html5.robot#L15

fetches the value right back.

At this point `${value}` variable is: `22019-02-11` resulting into `ValueError: year 22019 is out of range`

So, running the python code you just mentioned does work but the input data for parse is not.


* Input field type datetime-local

When setting datetime-local with the string "04-03-002019 01:45PM", Get Value from it gives back a string: "2019-03-04T01:45". This string does not have any indication if the time is PM or AM so AM is assumed and when that is then passed to `Parse Two Dates`, it will return `2019-03-04 01:45:00`.. As it should. 

* Input field type time

Same as previous datetime-local .. After setting the id:type field to "02:43PM", fetching its value is in format "02:34", which once again does not have indication if its AM or PM, and then calling parse() on these values will give wrong times. 


TL;DR:

Its not python-dateutil issue but browser behaviour and/or your expectations that setting time in american format will work for those input fields..

**Comment by aaltat** [comment]
So setting the date in `dd/mm/yyyy` format is not working for you but the `dd-mm-yyyy` format works.

And setting time in PM doesn't work for you, because your locale uses 24h clock.

Did I remember to say that I hate different date and time formats. 

Thanks for details and I am open to suggestions too.

**Comment by rasjani** [comment]
parse the string into platform specific time/date stamp first, insert it into element, and then get it to see that it was correctly set and retrieved ? dunno if that makes the test moot after that though..

**Comment by aaltat** [comment]
I was thinking something similar, but no idea how to actually do it. Any good pointers?

**Comment by aaltat** [comment]
Something like this might work: 
```
from datetime import datetime
date = datetime.today().strftime('%x %X')
print(date)
```

**Comment by aaltat** [comment]
I can reproduce this problem in my Windows machine. Will try to fix it.

**Comment by aaltat** [comment]
Locale is difficult problem to solve. Ended up by just checking that `${value}` is not empty.
